config/components.php

```diff
@@ -140,13 +141,21 @@
      * @param array $options Markdown options
      * @param bool $inline Whether to wrap the text in `<p>` tags (deprecated: set via $options['inline'] instead)
      * @return string
-     * @todo add deprecation warning for $inline parameter in 3.7.0
      * @todo remove $inline parameter in in 3.8.0
      */
-    'markdown' => function (App $kirby, string $text = null, array $options = [], bool $inline = false): string {
+    'markdown' => function (App $kirby, string $text = null, array $options = [], bool $inline = null): string {
         static $markdown;
         static $config;
 
+        // warning for deprecated fourth parameter
+        if ($inline === null) {
+            $inline = false;
+        } elseif (isset($options['inline']) === false) {
+            // @codeCoverageIgnoreStart
+            Helpers::deprecated('markdown component: the $inline parameter is deprecated and will be removed in Kirby 3.8.0. Use $options[\'inline\'] instead.');
+            // @codeCoverageIgnoreEnd
+        }
```

Comment:
If I understood correctly;

`$options['inline']` is required? I would expect `$inline !== null` check for deprecation. If it is not `null`, it means that the 4th parameter is actively being used.

Reply:
We need to continue passing $inline as 4th parameter ourselves in App::markdown() -- otherwise we could break custom components already.

In App::markdown() we pass it both as 4th parameter and as array key. That way in our default component we can throw a warning if 4th parameter is passed but not included as array key.

Reply:
I got it 👍